test/AdapterTests.cpp

```diff
@@ -371,27 +371,6 @@ TEST_CASE( "Test destination mode", "[destination]") {
     {
     return (msg.type() == com::amazonaws::iot::securedtunneling::Message_Type_STREAM_RESET) && msg.streamid() == 1;
     });
-    destination_socket.close();
```

Comment:
should we wait until we get acknowledgement from the destination LP that the stream is reset now before closing the connection? 

on which line did we see the failure?

Reply:
https://github.com/aws-samples/aws-iot-securetunneling-localproxy/blob/master/test/AdapterTests.cpp#L384
this line was originally causing failure

Reply:
I tried put `tcp_accept_thread.join();` before `REQUIRE( accepted );` and it stucks there, seems that `tcp_accept_thread.join();` never finished and join back

Reply:
and that is not consistent, sometimes it passes sometimes it sits there got stuck